### ISSUE TITLE
Using the render block filter on classic blocks returns null blockName which is incorrect

### ISSUE THREAD

**Issue by jaclyntan** [WordPress/gutenberg#14630]
**Describe the bug**
as described in #12862
Using the render_block filter returns a null blockName for the classic block

edit: sorry I thought it returned heading but actually, it returns nothing at all. There were a tonne of gutenberg specific tags in the HTML of the classic block I was editing which returned random haeding classes.
Regardless- the filter returns nothing for the classic block which doesn't seem right to me.

Using the following code:
```
add_filter( 'render_block', function( $block_content, $block ) {
    // Target core/* and core-embed/* blocks.
    if ( preg_match( '~^core/|core-embed/~', $block['blockName'] ) ) {
       $block_content = sprintf( '<div class="gb-block ' . sanitize_title($block['blockName']) . '">%s</div>', $block_content );
    }
    return $block_content;
}, PHP_INT_MAX - 1, 2 );
```

This returns a `core-heading` class which is incorrect as the classic block is not a heading block.

**To Reproduce**
1. Add content in a classic block on a post
2. Add the following code in functions.php
```
add_filter( 'render_block', function( $block_content, $block ) {
    // Target core/* and core-embed/* blocks.
    if ( preg_match( '~^core/|core-embed/~', $block['blockName'] ) ) {
       $block_content = sprintf( '<div class="gb-block ' . sanitize_title($block['blockName']) . '">%s</div>', $block_content );
    }
    return $block_content;
}, PHP_INT_MAX - 1, 2 );
```
3. Load the saved post with gutenberg edits and inspect element on the classic block content

**Expected behavior**
The class should be something obvious like 'core-classic'

**Additional context**
- Using WP 5.1.1
- I am currently using the render block filter to wrap blocks with classes for styling purposes, using the classic block to wrap paragraphs and headings and other media together. Ideally we should just have a container/wrapping block.

**Comment by swissspidy** [comment]
Worth noting that the Classic block is not technically a block. It's just a wrapper around a bunch of loose HTML tags in your post content. That's why there is no block name.

> Ideally we should just have a container/wrapping block.

That is in the works.

**Comment by gziolo** [comment]
I think it should be moved to Trac as the logic for the filter highlighted here is in WordPress core.

**Comment by loratov** [comment]
I'm getting a null value for the core/image block as well.

**Comment by spetrey** [comment]
Looks like all classic blocks have the blockName set to null for some reason. That doesn't seem very helpful.
Relevant: https://github.com/WordPress/WordPress/blob/master/wp-includes/blocks.php#L697

**Comment by kathrynwp** [comment]
@jaclyntan 

> Ideally we should just have a container/wrapping block.

There are now container divs around Classic blocks that can be targetted. I just tested and the container elements have classes starting with with `wp-container`.

Tested with:

WordPress 6.0.1
Twenty Twenty-Two 1.2
No plugins active OR just Gutenberg  13.6.0 active

Given that there are now container blocks, and that there's been no movement on this issue in over three years, I'm going to go ahead and close this out. If you still think a specific class should be added to Classic blocks, let me know and I'll see if I can get some other eyes on this.
